agent/agent.go

```diff
@@ -130,6 +131,55 @@ func (a *Agent) Run() []error {
 	return errs
 }
 
+// DryRun runs the agent to log what would occur during a run, without issuing any commands or writing to disk.
+func (a *Agent) DryRun() []error {
+	var errs []error
+
+	a.l.Info("Starting dry run")
+
+	// glob "*" here is to support copy/paste of seeker identifiers
+	// from -dryrun output into select/exclude filters
+	a.tmpDir = "*"
+	a.l.Info("Would copy included files", "includes", a.Config.Includes)
+
+	// Running healthchecks for products. We don't want to stop if any fail though.
+	a.l.Info("Checking product availability")
+	if errProductChecks := a.CheckAvailable(); errProductChecks != nil {
```

Comment:
I wonder if dryrun should just check whether the required clients are installed using the products' ClientCheck commands instead of also running a healthcheck. I could see it being a bit unexpected to a user if dryrun made any outbound calls.

I know this is existing behavior, so this PR might not be the place to change it, but seeing the `CheckAvailable` call here reminded me of it.

Reply:
I agree that issuing http reqs does break the contract a little bit, even if it's helpful. Maybe this is our first usecase for multiple commands in hcdiag?

I could see adding:

`hcdiag validate`
`hcdiag dryrun`
`hcdiag run`